### DISCUSSION TITLE
Can't get ZMK to build with Mouse Keys

### DISCUSSION THREAD

**Discussion by evernal** [manna-harbour/miryoku#208] [Q&A] [answered]
Working in my branch [here](https://github.com/evernal/miryoku_zmk) that's up to date with manna-harbour:master. Tried to follow the Mouse keys PR guidance at the bottom of the Readme with this [build-input job](https://github.com/evernal/miryoku_zmk/actions/runs/4708886824/jobs/8351613439), which throws errors (copied below) on merging ftc-zmk/mouse-ftc. Is this a known issue or am I missing something? There's another Q&A #170 from last November where it looks like it was working for someone else then at least. If not I guess I'll have to go do some diffs.

Run test "$MIRYOKU_DEBUG" = 'MIRYOKU_DEBUG_TRUE' && set -x
variables
outboards
zmk
merge
  From https://github.com/ftc/zmk
   * branch              mouse-ftc  -> FETCH_HEAD
   * [new branch]        mouse-ftc  -> ftc-zmk/mouse-ftc
  Auto-merging docs/sidebars.js
  Auto-merging app/src/main.c
  CONFLICT (content): Merge conflict in app/src/main.c
  Auto-merging app/src/hog.c
  Auto-merging app/src/hid_listener.c
  Auto-merging app/src/hid.c
  Auto-merging app/src/endpoints.c
  Auto-merging app/include/zmk/hid.h
  CONFLICT (content): Merge conflict in app/include/zmk/hid.h
  Auto-merging app/Kconfig
  Auto-merging app/CMakeLists.txt
  Automatic merge failed; fix conflicts and then commit the result.
  Error: Process completed with exit code 1.

**Comment by manna-harbour** [comment]
There was a recent breaking change in ZMK. Use `urob/zmk/mouse-3.2` instead of `ftc/zmk/mouse-ftc`.

Here's your above build: https://github.com/manna-harbour/miryoku_zmk/actions/runs/4712926361

**Reply by evernal** [reply]
Got it, thank you!

**Reply by manna-harbour** [reply]
Fixed in https://github.com/manna-harbour/miryoku_zmk/commit/e4a45f149e4a158e59e5343d2fa134107b8da931.

**Comment by ibeex** [comment]
hi, again there is conflict now with new branch `urob-zmk/mouse-3.2`
```
  From https://github.com/urob/zmk
   * branch              mouse-3.2  -> FETCH_HEAD
   * [new branch]        mouse-3.2  -> urob-zmk/mouse-3.2
  CONFLICT (add/add): Merge conflict in docs/docs/behaviors/mouse-emulation.md
  Auto-merging docs/docs/behaviors/mouse-emulation.md
  Auto-merging app/src/hog.c
  CONFLICT (content): Merge conflict in app/src/hog.c
  Auto-merging app/src/hid.c
  CONFLICT (content): Merge conflict in app/src/hid.c
  CONFLICT (add/add): Merge conflict in app/src/events/mouse_button_state_changed.c
  Auto-merging app/src/events/mouse_button_state_changed.c
  Auto-merging app/src/endpoints.c
  CONFLICT (content): Merge conflict in app/src/endpoints.c
  CONFLICT (add/add): Merge conflict in app/src/behaviors/behavior_mouse_key_press.c
  Auto-merging app/src/behaviors/behavior_mouse_key_press.c
  CONFLICT (add/add): Merge conflict in app/include/zmk/mouse.h
  Auto-merging app/include/zmk/mouse.h
  Auto-merging app/include/zmk/hog.h
  CONFLICT (content): Merge conflict in app/include/zmk/hog.h
  Auto-merging app/include/zmk/hid.h
  CONFLICT (content): Merge conflict in app/include/zmk/hid.h
  CONFLICT (add/add): Merge conflict in app/include/zmk/events/mouse_button_state_changed.h
  Auto-merging app/include/zmk/events/mouse_button_state_changed.h
  Auto-merging app/include/zmk/endpoints.h
  CONFLICT (content): Merge conflict in app/include/zmk/endpoints.h
  CONFLICT (add/add): Merge conflict in app/include/dt-bindings/zmk/mouse.h
  Auto-merging app/include/dt-bindings/zmk/mouse.h
  CONFLICT (add/add): Merge conflict in app/dts/behaviors/mouse_key_press.dtsi
  Auto-merging app/dts/behaviors/mouse_key_press.dtsi
  Auto-merging app/dts/behaviors.dtsi
  CONFLICT (content): Merge conflict in app/dts/behaviors.dtsi
  Auto-merging app/Kconfig
  CONFLICT (content): Merge conflict in app/Kconfig
  Auto-merging app/CMakeLists.txt
  CONFLICT (content): Merge conflict in app/CMakeLists.txt
  Automatic merge failed; fix conflicts and then commit the result.
  Error: Process completed with exit code 1.
```

**Reply by manna-harbour** [reply]
There is currently mouse related work going on in zmk main, so the mouse keys branch will need to be rebased again. You might find a new branch to use in the ZMK discord. In the mean time you can build from the mouse keys branch directly by using `urob/zmk/mouse-3.2` in the workflow `branches` field, but it will not include recent additions to zmk main.

**Reply by gbarkhatov** [reply]
@manna-harbour when zmk mouse work will be merged with their main branch, will there going to be an update to `miryoku` readme files?

**Reply by manna-harbour** [reply]
When a rebased mousekeys branch is available the docs will be updated, and when mousekeys are in ZMK main they will be enabled in Miryoku ZMK by default.

**Reply by manna-harbour** [reply]
Current status of mousekeys in ZMK is described at https://discord.com/channels/719497620560543766/845285481888743434/1177706631321104456.

**Comment by fsqonbdk** [comment]
Confirmed, merging only urob/zmk/mouse-3.2 compiles successfully. Thank you.

**Comment by manna-harbour** [comment]
Support for mouse buttons in ZMK main is now included by default, and `MIRYOKU_KLUDGE_MOUSEKEYSPR` has been updated for https://github.com/caksoylar/zmk/tree/feat/mouse-keys-3.2. See https://github.com/manna-harbour/miryoku_zmk#mouse-keys.

Support for the new mousekeys PR https://github.com/zmkfirmware/zmk/pull/2027 is also available. See https://github.com/zmkfirmware/zmk/pull/2027#issuecomment-1848392225.

**Comment by ddulic** [comment]
Heya, getting conflicts again

<details>
  <summary>Input</summary>

  ```
set-output()
  {
    json=`echo "$2" | jq -cnR 'inputs | split(",";"")'`
    echo "$1=$json" >> $GITHUB_OUTPUT
  }
  set-output "board" "nice_nano_v2"
  set-output "shield" "nice_nano_v2"   set-output "shield" "corne_left nice_view_adapter nice_view,corne_right nice_view_adapter nice_view"
  set-output "alphas" "default"
  set-output "nav" "default"
  set-output "clipboard" "Mac"
  set-output "layers" "default"
  set-output "mapping" "default"
  set-output "custom_config" "#define MIRYOKU_KLUDGE_MOUSEKEYSPR"
  set-output "kconfig" "default"
  set-output "branches" "zmkfirmware/zmk/main caksoylar/zmk/feat/mouse-keys-[3](https://github.com/ddulic/miryoku_zmk/actions/runs/8030549492/job/21938038761#step:2:3).2"
  ```
</details>

```
From https://github.com/caksoylar/zmk
   * branch              feat/mouse-keys-3.2 -> FETCH_HEAD
   * [new branch]        feat/mouse-keys-3.2 -> caksoylar-zmk/feat/mouse-keys-3.2
  Auto-merging docs/docs/intro.md
  Removing app/src/mouse.c
  Auto-merging app/src/main.c
  CONFLICT (content): Merge conflict in app/src/main.c
  Auto-merging app/src/hog.c
  Auto-merging app/Kconfig
  CONFLICT (content): Merge conflict in app/Kconfig
  Auto-merging app/CMakeLists.txt
  Automatic merge failed; fix conflicts and then commit the result.
  Error: Process completed with exit code 1.
```

**Comment by manna-harbour** [comment, marked as answer]
Mouse support in ZMK is in active development, so the mouse keys branch will periodically become out of date and need rebasing. Ask the branch maintainer for a rebase, or build from the mouse keys branch directly instead of attempting an automatic merge.

The docs have been updated. See https://github.com/manna-harbour/miryoku_zmk#mousekeys-pr and https://github.com/manna-harbour/miryoku_zmk#branches.

**Reply by manna-harbour** [reply]
The recommended mouse keys branch is now https://github.com/petejohanson/zmk/tree/feat/pointers-move-scroll, from PR https://github.com/zmkfirmware/zmk/pull/2027. Use `zmkfirmware/zmk/main petejohanson/zmk/feat/pointers-move-scroll` with workflow builds, or `petejohanson/zmk/feat/pointers-move-scroll` if there are merge conflicts.

Miryoku ZMK has now been updated. See https://github.com/manna-harbour/miryoku_zmk#mousekeys-pr. Here's a sample build run: https://github.com/manna-harbour/miryoku_zmk/actions/runs/8977387449.

**Reply by GrantCuster** [reply]
I'm able to build successfully with the new branch and flash it to a technikable@1.1. However after flashing the mouse keys do not work. If I hold down the mouse layer and use the mouse keys they just pass through as regular keys 'hjkl'. Everything else on the keyboard works well. Any tips for further debuging?

**Reply by manna-harbour** [reply]
The new branch requires Kconfig `CONFIG_ZMK_MOUSE=y` to be set manually. That was enabled by default in the old branch. Miryoku ZMK docs and example workflows have been updated now.
